scripts/exrrfs_ungrib.sh

```diff
@@ -39,31 +39,33 @@ elif [[ "${prefix}" == "GEFS" ]]; then
   ls -lth ${filea} ${fileb}
   cat ${filea} ${fileb} > GRIBFILE.AAA
 
-elif [[ "${prefix}" == "RAP" ]]; then
-  fstr=$(printf %02d ${FHRin})
-  GRIBFILE=${COMINrap}/rap.${CDATEin:0:8}/rap.t${CDATEin:8:2}z.wrfnatf${fstr}.grib2
-  ${cpreq} ${GRIBFILE} GRIBFILE.AAA
-
-elif [[ "${prefix}" == "RRFS" ]]; then
-  fstr=$(printf %02d ${FHRin})
-  GRIBFILE=${COMINrrfs1}/rrfs_a.${CDATEin:0:8}/${CDATEin:8:2}/rrfs.t${CDATEin:8:2}z.natlve.f${fstr}.grib2
-  ${cpreq} ${GRIBFILE} GRIBFILE.AAA
-
 else
-  echo "ungrib PREFIX=${prefix} not supported"
-  err_exit
+  echo "ungrib PREFIX=${prefix} from xml"
 fi
 #
 # generate the namelist on the fly
-CDATEout=$($NDATE ${FHR} ${CDATE})
-start_time=$(date -d "${CDATEout:0:8} ${CDATEout:8:2}" +%Y-%m-%d_%H:%M:%S) 
-end_time=${start_time}
-interval_seconds=3600
 if [[ "${MESH_NAME}" == "conus3km"   ]]; then
   dx=3; dy=3
 else
   dx=12; dy=12
 fi
+
+fhr_all=$(seq $((10#${OFFSET})) $((10#${INTERVAL})) $(( 10#${OFFSET} + 10#${LENGTH})) )
+for fhr in  ${fhr_all}; do
+
+  HHH=$(printf %03d ${fhr})
+  DATAH=${DATA}/${HHH}
+  mkdir -p ${DATAH}
```

Comment:
Per the NCO implementation standard, it is expected to create/prepare working directories in the J-job instead of the ex-script.

Reply:
removed this part.